pysmt/solvers/z3.py

```diff
@@ -418,7 +424,14 @@ def _back_single_term(self, expr, args, model=None):
                 return self.mgr._Algebraic(Numeral(expr))
             else:
                 # it must be a symbol
-                res = self.mgr.get_symbol(str(expr))
+                try:
+                    res = self.mgr.get_symbol(str(expr))
+                except UndefinedSymbolError:
+                    import warnings
+                    symb_type = self._z3_to_type(expr.sort())
+                    res = self.mgr.Symbol("__z3_%s" % str(expr), symb_type)
```

Comment:
Using FreshSymbol instead of symbol would be much better, because it GUARANTEES no name clashing